Add semantic field type documentation (#154174)
elastic/elasticsearch: docs/reference/query-languages/query-dsl/query-dsl-knn-query.md

      For an example request, refer to [`text_embedding`](#text-embedding-builder). For a broader overview of semantic kNN search, refer to [Perform semantic search](docs-content://solutions/search/vector/knn.md#knn-semantic-search).

$$$knn-query-builder-embedding$$$ `embedding` {applies_to}`stack: preview 9.4` {applies_to}`serverless: preview`
:   Build the query vector by generating an embedding from text, image, audio, video, or PDF input. This enables multimodal search, where different types of input can be used to generate a vector and retrieve similar documents. For an example, refer to [`embedding`](#embedding-builder).

      **Parameters for `embedding`**:

        `inference_id`

>         :   (Optional, string) The ID of the {{infer}} endpoint used to generate the embedding. It must reference an endpoint configured with the `embedding` task type. When the query targets only inference fields, Elasticsearch can infer this value from the field mapping(s). Specify it when querying a `dense_vector` field.

Reviewer: A note for the future: When we refer to "inference fields", we should link back to the common root docs for `semantic` and `semantic_text`. Those don't exist yet, but noting for the future.

Author: right, that's for follow up work

>         :   (Optional, string) The ID of the {{infer}} endpoint used to generate the embedding. It must reference an endpoint configured with the `embedding` task type. When the query targets only inference fields that use the same endpoint, Elasticsearch can infer this value from the field mapping. Specify it when querying a `dense_vector` field or fields that use different endpoints.

Reviewer: ```suggestion
        :   (Optional, string) The ID of the {{infer}} endpoint used to generate the embedding. It must reference an endpoint configured with the `embedding` task type. When the query targets only inference fields, Elasticsearch can infer this value from the field mapping(s). Specify it when querying a `dense_vector` field.
```
